course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,99 @@
+#include "unique_ptr.h"
+#include "unique_ptr.cc"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+template<template<class...> class UniquePtrType>
```

Comment:
The fixed pointed-to type was my first approach, and my code was exactly as you suggested above. However, in order to test operator->, I wanted to use a different type (struct, in my case), which is why I went for the template-template.